BlueDBOnDisk/src/main/java/org/bluedb/disk/file/ReadFileManager.java

```diff
@@ -131,7 +131,11 @@ protected BlueFileMetadata readMetadata(DataInputStream dis) throws IOException
 				dis.reset();
 				return null; // Legacy file without metadata header
 			}
-			return object instanceof BlueFileMetadata ? (BlueFileMetadata) object : null;
+			if (!(object instanceof  BlueFileMetadata)) {
+				dis.reset();
+				return null; // Legacy file without metadata header, somehow serialized into an object successfully

```

Comment:
Your comment here makes it seem crazy that this would happen. But wouldn't this happen on every legacy file? Oh, legacy files don't have a size at the beginning. It is just a single item without even a size at the beginning. So ya, this would be pretty crazy if it happened.

Reply:
Yeah, hopefully this never occurs but it is good to be sure 👍